### ISSUE TITLE
Implement Native Contracts

### ISSUE THREAD

**Issue by nathanhourt** [EOSIO/eos#15]
Implement the Native Contracts in C++

- System Contract
  - [x] Create Account (name, owner authority, active authority)
  - [ ] Update Permission (account, permissionname, parent, authority)
  - [ ] Update Script (account, new script)
- Eos Contract
  - [x] Transfer (to, amount, memo)
  - [ ] TransferToLocked (to, amount)
- Staked Balance Contract
  - [x] Create Producer (owner name, signing key)
  - [x] Update Producer (account, signingkey)
  - [ ] Producer Vote Update (account, producer, does_approve)
  - [ ] Register As Foundation (account)
  - [ ] Foundation Vote Update (account, foundation, does_approve)
  - [ ] StartUnlockEOS (account, amount)
  - [ ] ClaimUnlockedEOS (account, amount)

**Comment by nathanhourt** [comment]
Rewrote issue to reflect that the "system contract" has now been broken up into three different contracts, each responsible for different things, collectively called "the native contracts"

**Comment by bytemaster** [comment]
I think we can merge all of these into a single "eos" contract that have different scope requirements:

For example,  `eos::transfer(from,to,amount)` only requires scope of from and to where as `eos::lock(from,to,amount)` requires scope of `from`, `to`, and `eos`.  

Under this model we could remove some of the 'notify' handlers and complexity associated with having some logic enforced in 'eos' and other logic enforced in 'system' and still more in 'staked'.  

In other words, via scope requirements we can merge all of these into a single contract while still having parallelism.

**Comment by heifner** [comment]
This is all OBE.
